### PR TITLE
Use zend_string macros to make it compile against PHP 7.3-dev

### PR DESCRIPTION
**Pull request by fmk**
The default branch was showing warnings when compiling with PHP 7.3-dev and the session handler was no working. Adding ZSTR_VAL() and ZSTR_LEN() macros in 3 lines fixed the problems.

### CODE REVIEW

**Comment by yatsukhnenko** [comment]
@fmk thanks! But this change isn't compatible with all previous versions of PHP.

**Comment by fmk** [comment]
I have tested this with PHP 7.1, 7.2 and 7.3-dev and they all work. What are the errors?

**Comment by fmk** [comment]
I swapped the logic for checking the version. It should now compile with version 5x and 7.x.

File: redis_session.c [hunk 1]
```diff
@@ -231,7 +231,11 @@ PS_OPEN_FUNC(redis)
             if (url->query != NULL) {
                 array_init(&params);
 
+#if (PHP_MAJOR_VERSION == 7) && (PHP_MINOR_VERSION > 2)
```
Comment (remicollet) [inline redis_session.c]: IMHO, using PHP_VERSION_ID is simpler (and will take care of PHP 8)

Comment (yatsukhnenko) [reply redis_session.c]: Agree with @remicollet , something like `#if (PHP_VERSION_ID < 70300)` would be much better

**Comment by yatsukhnenko** [comment]
@fmk Could you modify your PR to use PHP_VERSION_ID and squash commits? I'm going to merge this and release 3.1.5 as stable on PECL site.

**Comment by yatsukhnenko** [comment]
I've made changes by myself. Thanks!

**Pull request closed without merging**